code/modules/ninja/suit/suit.dm

```diff
@@ -82,7 +82,9 @@ Contents:
 	qdel(n_gloves)
 	qdel(n_shoes)
 	qdel(src)
-
+	if(energyKatana)
+		energyKatana.visible_message("<span class='warning'>[src] flares and then turns to dust!</span>")
+		qdel(energyKatana)
```

Comment:
Will this work after you qdel ninja himself?

Reply:
actually this should definitely go before qdeling src